### DISCUSSION TITLE
Traceroute

### DISCUSSION THREAD

**Discussion by dheryta** [openthread/openthread#8396] [Q&A] [answered]
Does trace route work with OpenThread?

**Comment by jwhui** [comment, marked as answer]
Traceroute will probably not work as you might expect. Traceroute relies on ICMPv6 Time Exceeded errors to have intermediate routers along the way generate IPv6 messages back to the original source. Thread forwards messages at a layer below IP, using the Mesh Addressing header from RFC 4944. As a result, delivering an IPv6 message across a set of Thread routers appears as a single IP hop even though multiple Thread routers may have participated in forwarding a given IP message.

**Comment by dheryta** [comment]
Sure.
That is what I expected.
Is there any utility that will ease the process of knowing the route?

On Mon, Nov 14, 2022 at 8:30 PM Jonathan Hui ***@***.***>
wrote:

> Traceroute will probably not work as you might expect. Traceroute relies
> on ICMPv6 Time Exceeded errors to have intermediate routers along the way
> generate IPv6 messages back to the original source. Thread forwards
> messages at a layer below IP, using the Mesh Addressing header from RFC
> 4944. As a result, delivering an IPv6 message across a set of Thread
> routers appears as a single IP hop even though multiple Thread routers may
> have participated in forwarding a given IP message.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openthread/openthread/discussions/8396#discussioncomment-4142361>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABLSMKK63L6SK3WSJQIITTLWILYUPANCNFSM6AAAAAASAN3V5E>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
--
Regards

Dheryta

**Reply by jwhui** [reply]
Take a look at:
- https://github.com/openthread/openthread/pull/8682
